client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -1906,14 +1918,20 @@
                 /* Creates a copy of dateline object from item.__proto__.dateline */
                 scope.$watch('item', function(item) {
                     if (item) {
-                        var updates = {'dateline': {'source': item.dateline.source, 'date': item.dateline.date,
-                            'located': item.dateline.located, 'text': item.dateline.text}};
-
-                        if (item.dateline.located) {
-                            var monthAndDay = $filter('formatDatelineToMMDD')(item.dateline.date, item.dateline.located);
-                            scope.datelineMonth = monthAndDay.month;
-                            scope.datelineDay = monthAndDay.day;
-                            scope.resetNumberOfDays(false);
+                        var updates = {dateline: {}};
+                        if (item.dateline) {
+                            var fields = ['source', 'date', 'located', 'text'];
+                            for (var i = 0; i < fields.length; i++) {
+                                if (item.dateline[fields[i]]) {
+                                    updates.dateline[fields[i]] = item.dateline[fields[i]];
+                                }
+                            }
+                            if (item.dateline.located) {
+                                var monthAndDay = $filter('formatDatelineToMMDD')(item.dateline.date, item.dateline.located);
+                                scope.datelineMonth = monthAndDay.month;
+                                scope.datelineDay = monthAndDay.day;
+                                scope.resetNumberOfDays(false);
+                            }
                         }
 
                         _.extend(item, updates);
```

Comment:
Do we have the tests covering this `$watch` handler in place, including the new code parts?